src/NuGet.Clients/PackageManagement.UI/PackageLoadContext.cs

```diff
@@ -25,16 +28,20 @@ internal class PackageLoadContext
 
         public IEnumerable<IVsPackageManagerProvider> PackageManagerProviders { get; private set; }
 
+        public PackageSearchMetadataCache CachedUpdates { get; private set; }
```

Comment:
Let's make it mutable and get rid off passing null in constructor.
Also I'd prefer renaming it to a generic CachedPackages.